solutions/ceoi-12-JobScheduling.mdx

```diff
@@ -0,0 +1,107 @@
+---
+id: ceoi-12-JobScheduling
+source: CEOI 2012
+title: Job Scheduling
+author: Chuyang Wang
+---
+
+We can use binary search to easily find the minimum number of machines needed for finishing the jobs within the given deadline. 
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N \log N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+#define all(x) begin(x), end(x)
+#define mp make_pair
+
+using pi = pair<int, int>;
+using vi = vector<int>;
+
+// test if it is possible to finish the jobs using given # of machines
+// return: first: possible or not, second: if possible, the schedule for the jobs
+pair<bool, vector<vi>> isFeasible(const vector<pi> &jobs, int N, int D, int machineCount)
+{
+  vector<vi> res(N);
+  int reqNum = 0;
+  for (int day = 1; day <= N; day++)
+  {
+    for (int j = 0; j < machineCount; j++)
+    {
+      // if all jobs before and on this day are finished,
+      // we can go to the next day, even if there are usable machines left
+      if (jobs.at(reqNum).first > day)
+        break;
+			
+      // if the current date is before the deadline for the job
+      if (jobs.at(reqNum).first + D >= day)
+        res[day - 1].push_back(jobs.at(reqNum++).second);
```

Comment:
brackets?

Reply:
doesn't matter

Reply:
I've read somewhere that vector::at will provide nicer error message, but for consistency, ya, I will change it to [].